### DISCUSSION TITLE
Repository reliability for big projects in the future

### DISCUSSION THREAD

**Discussion by HamedSiaban** [DuCanhGH/next-pwa#80] [Q&A]
Hello.
I was researching to find the best pwa supporting solution for a nextjs project that supports the new nexjs 13 `app directory` feature.
And I think I'm in the right place. Because there are a lot of forked repos by people but I think yours is the most improved and popular one.
Well done. :+1: 

So I'm starting to implement a huge project for a company and I want to make sure that I can rely on this repository so I wouldn't have to submit to a huge and time wasting migration of all my work to another framework or library in the future.

I also have another option: To use `create-react-app` and opt to `CSR` and use it's pwa plugin. It will have it's cons business-wise but it's a trade off for me. 

Can you share your thoughts with me?
Thanks :pray:

**Comment by DuCanhGH** [comment]
Well if you're gonna use `create-react-app` with its SW stuff I'd recommend using Vite with `vite-plugin-pwa` instead as CRA hasn't been maintained for ages. As for this project though, it will remain maintained as long as there's demand (the opposite happens when, for example, Next releases an official PWA plugin) and I can keep doing it. For now, it is pretty much guaranteed to get at least an update a month. You can also fork the project and periodically update it to ensure that even when something happens you will still be able to access it :)

**Reply by omerman** [reply]
You are the best @DuCanhGH ❤️

**Reply by HamedSiaban** [reply]
@DuCanhGH Thanks for the great explanation.
In the end I decided to use your repository and so far I had no problem working with it.
Hopefully it would be a great experience.
:pray:
